punch/cli.py

```diff
@@ -152,21 +159,37 @@ def main(original_args=None):
     current_version = ver.Version.from_file(args.version_file, config.version)
     new_version = current_version.copy()
 
+    if args.part:
+        args.action = "punch:increase"
+        args.action_options = "part={}".format(args.part)
+
+    if args.set_part:
+        args.action = "punch:set"
+        args.action_options = args.set_part
+
     if args.action:
-        action_dict = config.actions[args.action]
-        action = act.Action.from_dict(action_dict)
-        new_version = action.process_version(new_version)
-    else:
-        if args.part:
-            new_version.inc(args.part)
+        try:
+            action_dict = config.actions[args.action]
+        except KeyError:
+            print(
+                "The requested action {} is not defined.".format(
+                    args.action
+                )
+            )
+            sys.exit(0)
 
-        if args.set_part:
-            if args.reset_on_set:
-                part, value = args.set_part.split('=')
-                new_version.set_and_reset(part, value)
-            else:
-                set_dict = dict(i.split('=') for i in args.set_part.split(','))
-                new_version.set(set_dict)
+        try:
+            action_name = action_dict.pop('type')
```

Comment:
This is good. It does make me think about how punch can be extended; in the future it would be good to load some extension interface before checking this, so users could add their own options locally without patching punch directly.

But for now this is just fine 👍

Reply:
I agree, it looks very limited. Let's merge this for the moment, it still doesn't contain all your suggestions. Feel free as usual to submit ideas and/or code. Thanks!